app/src/battery.c

```diff
@@ -18,12 +19,16 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 #include <zmk/battery.h>
 #include <zmk/events/battery_state_changed.h>
 
-const struct device *battery;
-
 static uint8_t last_state_of_charge = 0;
 
 uint8_t zmk_battery_state_of_charge() { return last_state_of_charge; }
 
+#if DT_HAS_CHOSEN(zmk_battery)
+static const struct device *const battery = DEVICE_DT_GET(DT_CHOSEN(zmk_battery));
+#else
+static const struct device *battery;
```

Comment:
Can we add a small warn here:

```suggestion
#warning "Using specially labeled BATTERY node is deprecated, in favor of setting a zmk,battery chosen node."
static const struct device *battery;
```

Reply:
Only if you can either find a macro to get a device by label at compile time (I couldn't) or are okay with that warning appearing for every keyboard that doesn't have a battery sensor at all.

Reply:
We only compile `battery.c` if ZMK_BLE is enabled:

```
target_sources_ifdef(CONFIG_ZMK_BLE app PRIVATE src/battery.c)
```

So I think that's reasonable/safe for now?

Reply:
That means you probably have a battery, but it doesn't necessarily mean your hardware has a way to *check* the battery. Do you think that scenario is unlikely that it's okay to give a meaningless warning in it?

Reply:
I think we *do* warn here, because I think our *next* step is to do something like:

```
DT_CHOSEN_ZMK_BATTERY := zmk,battery

config ZMK_BATTERY_READ
    bool "Read battery levels to report them"
    default $(dt_chosen_enabled,DT_CHOSEN_ZMK_BATTERY)
```

Then use that Kconfig in CMakeLists.txt to determine if we build this.

Reply:
Done.